lib/cosmos/core_ext/time.rb

```diff
@@ -211,10 +211,12 @@ def seconds_of_day
 
   # @return [String] Date formatted as YYYY/MM/DD HH:MM:SS.US
   def formatted(include_year = true, fractional_digits = 3)
+    fractional = ''
+    fractional = self.strftime(".%#{fractional_digits}N") if fractional_digits > 0
```

Comment:
I don't really want two calls to strftime, that will be considerably slower.   Can you just sprintf the microseconds value from self?